tsdb/db.go

```diff
@@ -194,6 +195,55 @@ func VerifyChainPathExists(chainPath string) bool {
 	return os.IsExist(err)
 }
 
+// EXPERIMENTAL (ref: https://github.com/bench-routes/bench-routes/issues/242)
+func (c *Chain) addNullBlock() {
+	chainReadOnly := ReadOnly(c.Path).Refresh()
+	bstream := chainReadOnly.BlockStream()
+
+	blockStream := *bstream
+
+	lengthBlockStream := len(blockStream)
+
+	if lengthBlockStream == 0 {
+		return
+	}
+
+	block := blockStream[lengthBlockStream-1] // getting the most recent block added to the disk
+
+	blockNormTime := block.GetNormalizedTime()
+	curNormTime := GetNormalizedTimeCalc()
+	blockType := block.Type
+
+	if blockType != "sys" && blockType != "journal" {
+		return
+	}
+
+	var blockChain []Block // Null blocks to be added
+	sub := 0
+	time := int64(time.Second * 5) // time is in nano seconds
+
+	/*
+		Starting from the current normalized time, we go back
+		and add null blocks until we get past the block
+		normalized time which is the most recent block added
+		at the time when bench-routes was stopped
+	*/
+	for curNormTime > blockNormTime {
+		b := GetNewBlock(blockType, "null")
+		b.Timestamp = CalcTimeStamp(sub)
+		b.NormalizedTime = curNormTime
+		blockChain = append(blockChain, *b)
+		sub = sub + int(time/1000000000)
+		curNormTime -= time
+	}
```

Comment:
Aren't we still adding so many null blocks? The idea was to just add a single null block only and not so many null blocks. the UI will consider the null block and apexcharts will automatically remove the graph until the next value (which will be immediately after this null block).

imagine time of say -> 1, 2, 3, 4, 5, 11, 12, 13, 14

then, we do

1, 2, 3, 4, 5, null, 11, 12, 13, 14

Reply:
Thanks for the review @Harkishen-Singh. I have made the changes. Kindly check